Tasks/JenkinsQueueJobV2/job.ts

```diff
@@ -131,6 +131,14 @@ export class Job {
         this.working = false;
     }
 
+    
+    private RetryConnection(): void {
+        const thisJob: Job = this;
+        thisJob.queue.TaskOptions.retryNumber--;
```

Comment:
I'm not sure if it's good idea to change it in TaskOptions - I would rather treat is as immutable input parameters of the task... Could we keep it in separate variable somewhere?